docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
 
-## Design & implementation
+## Design
+
+### Overview of GameFlow
+![GameFlow.jpg](UML%20diagram%2FGameFlow.jpg)
+
+### Architecture
+
+![Architecture.png](UML%20diagram%2FArchitecture.png)
+
+The **Architecture diagram** above showcases the high-level design of the EconoCraft Pro application. 
+
+The following section would give a brief overview of the different components of the application and their interactions
+with each other.
+
+#### Main components:
+
+`Main` (consisting of class `EconoCraft` and `EconoCraftLogic`) 
+is in charge of the game initialization and starting the main game loop.
+
+* At game initialization, it reads the necessary information from prompted user input and develops the player profile.
+* When game starts, it initializes the game logic and starts the main game loop.
+
+The functionality of the game is divided into different components, each responsible for a different aspect of the game.
+
+* `PlayerProfile`: Represents the player's profile and is responsible for storing and updating the player's information.
+* `Parser`: Responsible for parsing the user input.
+* `ResponseManager`: Responsible for generating the response to the user input.
+* `EconoCraftLogic`: Responsible for executing user command and updating the game state.
+* `MiniGame`: Responsible for handling the mini-games that the player can play to gain rewards.
+* `CommandFactory`: Responsible for using the parsed user input to produce executable commands.
+
+#### Component interaction:
+
+The *Sequence Diagram* below showcases the interaction between the different components of the EconoCraft Pro 
+application when a user inputs a command `work`.
+
+For `CommandFactory` and `Minigame`,
+* Each defines its API for creating commands and mini-games respectively, where
+  `Command` is the API for `CommandFactory` and `MiniGame` is the API for `MiniGame`.
+* Implements its functionality with concrete classes such as `WorkCommand` and `TypingGame`.
+
+The sections below would give more details of each component.
+
+## UI components
+
+UI consists of the following components:
+* `Parser`
+* `ResponseManager`
+* `CommandType`
+
+![UI.png](UML%20diagram%2FUI.png)
+
+The **UI components**,
+* `Parser` parses the user input for `CommandFactory` to produce `Command`
+* `ResponseManager` generate the response to the user according to the *command execution* and *game logic*.
+
+## EconoCraftLogic components
+
+Here is the partial class diagram of the `EconoCraftLogic` component:
+
+![Logic.png](UML%20diagram%2FLogic.png)
+
+> [!NOTE]
+> - The `XYZ` in `XYZCommand` represents the exact command e.g., `WorkCommand`, `RestCommand`, `ExerciseCommand`.
+> - The `XYZ` in `XYZMiniGame` represents the exact mini-game e.g., `TypingGame`, `TicTacToe`, etc.
+> - The `XYZ` in `StockXYZ` represents the stock label e.g., `StockOne`, `StockTwo`, etc.
+
+Here is the sequence diagram of the `EconoCraftLogic` executing the command:
+
```

Comment:
Notation seems to be compliant with the notation covered in the course.
The diagram is suitable for the purpose it is used for and is not too complicated. Well Done!